### DISCUSSION TITLE
osu! with ProMotion on iPhone hits 120fps but dropped after few seconds if no touch input

### DISCUSSION THREAD

**Discussion by Naozumi520** [ppy/osu#19980] [Q&A] [answered]
In discussion #15721, we confirmed that osu! is not running at 120fps and added ProMotion support soon. 
However in my usual play, it rarely hits 120fps. When you selete a map and start playing, it hits 120fps but soon dropped to 60fps or lower. Frame limiter or Threading mode does not affect the result.

I'm playing with an iPhone 13 Pro. Since it is a mobile platform, I have also considered whether it is a performance problem, but even if some other stressful games have no performance problems on this phone, performance should not be a problem, especially with the A15 chip. Thermal throttling is also not the possible, phone has no apparent heat in the test, and I just started osu! in a minute.

Here's a video of the actual behavior and with different settings:
https://www.youtube.com/watch?v=6IzRVEyaaXE

**Comment by peppy** [comment, marked as answer]
The only settings you should use are:

- vsync
- single thread

If the game falls below 120fps, that's because of vsync (a single frame taking too long is enough to throw it off). To improve on this, we need to migrate to metal, which will happen eventually.

**Reply by Naozumi520** [reply]
Hey peppy, I just found out that the reason of the game falls to 60fps is because no touch input is provided. I just noticed if my finger leaved the screen like watching auto playing, the fps will falls to 60fps (I guess that's a ProMotion feature?) However, when my finger keep dragging on the screen while the fps falls, it will soon go back to 120fps. And I assuming this is a abnormal behavior?

**Reply by peppy** [reply]
That's likely the case. There's documentation of this (see https://github.com/ppy/osu/issues/9388 as one example) across the internet. Probably something you'll just have to put up with for now.
